docs/root/configuration/other_protocols/thrift_filters/payload_to_metadata_filter.rst

```diff
@@ -0,0 +1,95 @@
+<envoy_v3_api_field_extensions.filters.network.thrift_proxy.filters.payload_to_metadata.v3.PayloadToMetadata.FieldSelector.id>`
+for matching. The :ref:`field_selector
+<envoy_v3_api_field_extensions.filters.network.thrift_proxy.filters.payload_to_metadata.v3.PayloadToMetadata.Rule.field_selector>`
+is tied to a payload field when the linked list corresponds to a downward path which rooted in the top-level of the
+request message structure. :ref:`on_present
+<envoy_v3_api_field_extensions.filters.network.thrift_proxy.filters.payload_to_metadata.v3.PayloadToMetadata.Rule.on_present>`
+is triggered when corresponding the payload is present. Otherwise, :ref:`on_missing
+<envoy_v3_api_field_extensions.filters.network.thrift_proxy.filters.payload_to_metadata.v3.PayloadToMetadata.Rule.on_missing>`
+is triggered.
+
+Note that if the corresponding payload for a :ref:`rule
+<envoy_v3_api_msg_extensions.filters.network.thrift_proxy.filters.payload_to_metadata.v3.PayloadToMetadata.Rule>`
+is present but :ref:`on_present
+<envoy_v3_api_field_extensions.filters.network.thrift_proxy.filters.payload_to_metadata.v3.PayloadToMetadata.Rule.on_present>`
+is missing, no metadata is added for this :ref:`rule
+<envoy_v3_api_msg_extensions.filters.network.thrift_proxy.filters.payload_to_metadata.v3.PayloadToMetadata.Rule>`.
+. If the corresponding payload for a :ref:`rule
+<envoy_v3_api_msg_extensions.filters.network.thrift_proxy.filters.payload_to_metadata.v3.PayloadToMetadata.Rule>`
+is an empty string, neither :ref:`on_present
+<envoy_v3_api_field_extensions.filters.network.thrift_proxy.filters.payload_to_metadata.v3.PayloadToMetadata.Rule.on_present>`
+nor :ref:`on_missing
+<envoy_v3_api_field_extensions.filters.network.thrift_proxy.filters.payload_to_metadata.v3.PayloadToMetadata.Rule.on_missing>`
+is triggered. i.e., no metadata is added for this :ref:`rule
+<envoy_v3_api_msg_extensions.filters.network.thrift_proxy.filters.payload_to_metadata.v3.PayloadToMetadata.Rule>`.
+
+Currently payload to metadata filter doesn't support container type payload, i.e., list, set and map.
+
+This filter is designed to support payload passthrough. Performing payload to metadata filter
+can do deserialization once, and pass the metadata to other filters. This means that load balancing
+decisions, consumed from log and routing could all use payload information with a single parse.
+Also notably performing the parsing in payload passthrough buffer will mean deserialization once
+and not re-serializing, which is the most performant outcome.
+
+If any of the filter chain doesn't support payload passthrough, a customized non-passthrough
+filter to setup metadata is encouraged from point of performance view.
+
+A typical use case for this filter is to dynamically match requests with load balancer
+subsets. For this, a given payload field's value would be extracted and attached to the request
+as dynamic metadata which would then be used to match a subset of endpoints.
```

Comment:
Can you move this paragraph up to the start of the document? I think it's the first thing people would want to read. Also, could you add a sentence saying why would you need payload-to-metadata instead of header-to-metadata (I think the answer is if you are not using THeader)?

Reply:
Sure, will do!
Missing THeader in framed transport is one reason. Another reason is that we can not trust the downstream service always copy the field to the THeader correctly, and we want single truth of source.